ocpmodels/trainers/forces_trainer.py

```diff
@@ -372,25 +379,33 @@ def validate(self, split="val", epoch=None):
     def validate_relaxation(self, split="val", epoch=None):
         print("### Evaluating ML-relaxation")
         self.model.eval()
-        metrics = {}
-        meter = Meter(split=split)
-
-        mae_energy, mae_structure = relax_eval(
-            trainer=self,
-            traj_dir=self.config["task"]["relaxation_dir"],
-            metric=self.config["task"]["metric"],
-            steps=self.config["task"].get("relaxation_steps", 300),
-            fmax=self.config["task"].get("relaxation_fmax", 0.01),
-            results_dir=self.config["cmd"]["results_dir"],
-        )
 
-        metrics[
-            "relaxed_energy/{}".format(self.config["task"]["metric"])
-        ] = mae_energy
+        meter = Meter(split=split)
 
-        metrics[
-            "relaxed_structure/{}".format(self.config["task"]["metric"])
-        ] = mae_structure
+        # only supports batch_size = 1
+        # TODO: Batch relaxation support
+        for i, batch in enumerate(self.relax_loader):
+
+            mae_energy, mae_structure = relax_eval(
+                batch=batch,
+                model=self,
+                metric=self.config["task"]["metric"],
+                steps=self.config["task"].get("relaxation_steps", 300),
+                fmax=self.config["task"].get("relaxation_fmax", 0.01),
+                results_dir=self.config["cmd"]["results_dir"],
+                relax_opt=self.config["relax_opt"],
+            )
+            metrics = {
+                "relaxed_energy/{}".format(
+                    self.config["task"]["metric"]
+                ): mae_energy,
+                "relaxed_pos/{}".format(
+                    self.config["task"]["metric"]
+                ): mae_structure,
+            }
+
+            meter_update_dict = {}
+            meter.update(meter_update_dict)
 
         meter.update(metrics)
```

Comment:
Lines 407-410 are incorrect: 407-408 are not updating the metrics at all. And line 410 is only updating the metrics for the last batch.